doc-site/docs/tutorials/private-storage.md

```diff
@@ -0,0 +1,147 @@
+
+The Solidity contract remains the same as in the [Public Storage Tutorial](https://github.com/LF-Decentralized-Trust-labs/paladin/blob/main/solidity/contracts/tutorial/Storage.sol). However, the interaction is scoped to the privacy group.
+
+---
+
+## Step 1: Create a Privacy Group
+
+To enable private interactions, start by creating a privacy group with selected members.
+
+```typescript
+// Create a privacy group with Node1 and Node2
+logger.log("Creating a privacy group for Node1 and Node2...");
+const penteFactory = new PenteFactory(paladinNode1, "pente");
+const memberPrivacyGroup = await penteFactory.newPrivacyGroup(verifierNode1, {
+  group: {
+    salt: newGroupSalt(), // Generate a unique salt for the group
+    members: [verifierNode1, verifierNode2], // Include Node1 and Node2 as members
+  },
+  evmVersion: "shanghai",
+  endorsementType: "group_scoped_identities",
+  externalCallsEnabled: true,
+});
+
+if (!checkDeploy(memberPrivacyGroup)) {
+  logger.error("Failed to create the privacy group.");
+  return false;
+}
+logger.log("Privacy group created successfully!");
+```
+
+---
+
+## Step 2: Deploy the Contract in the Privacy Group
+
+Deploy the `Storage` contract within the created privacy group.
+
+```typescript
+logger.log("Deploying a private Storage contract...");
+const contractAddress = await memberPrivacyGroup.deploy(
+  storageJson.abi, // ABI of the Storage contract
+  storageJson.bytecode, // Bytecode of the Storage contract
+  verifierNode1 // Deploying as Node1
+);
+
+if (!contractAddress) {
+  logger.error("Failed to deploy the private Storage contract.");
+  return false;
+}
+logger.log(`Private Storage contract deployed! Address: ${contractAddress}`);
+```
+
+---
+
+## Step 3: Store and Retrieve Values as Group Members
+
+### Store a Value
+
+Group members can store values securely in the private contract.
+
+```typescript
+const privateStorage = new PrivateStorage(memberPrivacyGroup, contractAddress);
+
+logger.log("Storing value (125) in the private Storage contract...");
+const storeTx = await privateStorage.invoke(verifierNode1, "store", { num: 125 });
+logger.log("Value stored successfully! Transaction hash:", storeTx?.transactionHash);
+```
+
+---
+
+### Retrieve the Value as a Member
+
+Authorized group members can retrieve the stored value.
+
+```typescript
+logger.log("Node1 retrieving the stored value...");
+const retrievedValueNode1 = await privateStorage.call(verifierNode1, "retrieve", []);
+logger.log("Node1 retrieved the value successfully:", retrievedValueNode1["0"]);
+
+logger.log("Node2 retrieving the stored value...");
+const retrievedValueNode2 = await privateStorage
+  .using(paladinNode2)
+  .call(verifierNode2, "retrieve", []);
+logger.log("Node2 retrieved the value successfully:", retrievedValueNode2["0"]);
+```
+
+---
+
+## Step 4: Verify Privacy by Testing Unauthorized Access
+
+When an outsider (Node3) tries to access the private contract, the attempt should fail.
```

Comment:
This is abstracting away a lot of what privacy groups mean. I wonder if we need some more words about _why_ privacy groups work (because the private state information for these transactions was not distributed to this other node).

Reply:
It's not so much that they don't "have access to the contract". It's that all the inputs and outputs of each private transaction are shared privately, so they cannot reconstruct the current state of anything in that privacy group.